mypy/meet.py

```diff
@@ -34,7 +39,7 @@ def narrow_declared_type(declared: Type, narrowed: Type) -> Type:
     if isinstance(declared, UnionType):
         return UnionType.make_simplified_union([narrow_declared_type(x, narrowed)
                                                 for x in declared.relevant_items()])
-    elif not is_overlapping_types(declared, narrowed, use_promotions=True):
+    elif not is_overlapping_types(declared, narrowed):
```

Comment:
Do you have a test for this, or you jut noticed the inconsistency?

Reply:
Oh, what happened here is that that I changed the behavior of `is_overlapping_types` so it does use promotions by default. So, removing this param isn't actually causing any difference in behavior.